### ISSUE TITLE
Allow disabling auto-target on a cluster member

### ISSUE THREAD

**Issue by stgraber** [lxc/lxd#7948]
By default when LXD launches an instance without a `--target` passed, it will pick the cluster member with the least instances.

This works fine except that sometimes there are specialized cluster members with very different specifications which should not automatically get instances.

For this, we should introduce the ability to set config keys on cluster members and then introduce an initial key to disable automatic targeting on that member.

This would include:
 - Add DB table
 - Expand API struct
 - Add config key
 - Tweak scheduler to respect it
 - Update docs

**Comment by DDEFISHER** [comment]
I am going to work on this if that is okay.

**Comment by stgraber** [comment]
Absolutely! Assigned to you now.

**Comment by stgraber** [comment]
@DDEFISHER still looking into this?

**Comment by DDEFISHER** [comment]
@stgraber Yes I got side tracked with some other things but plan on looking into this still. If it's urgent though I understand if you need to re-assign.

**Comment by stgraber** [comment]
Nope, not urgent at present. Let me know if you have any question or need any help!

**Comment by stgraber** [comment]
@DDEFISHER hey there, just checking in to see if you're still planning on implementing this one.

**Comment by DDEFISHER** [comment]
Yes sorry about procrastinating on this. I am going to give myself until the end of this weekend to have something or I will un-assign myself if that works for you.

**Comment by stgraber** [comment]
yep, no problem :)

**Comment by DDEFISHER** [comment]
I was actually able to spend some time looking at this during the weekend. It was mostly spent refreshing myself with the project since it had been awhile but anyways I will keep myself assigned and continue working on this.

**Comment by stgraber** [comment]
Re-assigning to @masnax as we have this on roadmap and so need it done by end of month.

@masnax this should add a new `nodes_config` table with the usual key=value scheme.
This should support our usual `user.*` keys that can be set to any value the user wishes and have it exposed through GET/PUT/PATCH on /1.0/cluster/members/NAME.

We'll then add a `scheduler.manual` config key (better names welcome) which when set to true will exclude that cluster member from any automatic targeting, effectively only taking workloads if `--target` is directly used during a `lxc init` or `lxc launch` operation.

**Comment by tomponline** [comment]
@stgraber is this the thing we chatted about the other day re. cluster group targetting?

If so, did we decide on using a key that would accept a string value such as `scheduler.mode=` and then have `manual` and `auto`, and later we were going to add `targetted` which would include any groups that the member belonged to?

Also, is this only going to affect instance scheduling? If so is it worth calling the config key `scheduler.instance.x` to indicate thats what it influences?

**Comment by tomponline** [comment]
How about scheduler.instance with initial values of 'all' (default) and 'manual'. Then later can support 'group'.

**Comment by stgraber** [comment]
Yeah, that'd work.
